File: UseCases/tests/codes/seanhenry/mockgenerator/util/StringDecoratorTests.java
package codes.seanhenry.mockgenerator.util;

import junit.framework.TestCase;

public class StringDecoratorTests extends TestCase {

  class EmptyStringDecorator extends StringDecorator {

    public EmptyStringDecorator(StringDecorator decorator) {
      super(decorator);
    }

    @Override
    protected String decorate(String string) {
      return string;
    }
  }

  public void testShouldProcessStringInDecorator() {
    PrependStringDecorator decorator = new PrependStringDecorator(null, "prefix");
    StringDecorator stringDecorator = new EmptyStringDecorator(decorator);
    assertEquals("prefixName", stringDecorator.process("name"));
  }

  public void testShouldNotGiveStringToNullDecorator() {
    StringDecorator stringDecorator = new EmptyStringDecorator(null);
    assertEquals("name", stringDecorator.process("name"));
  }

  public void testShouldProcessStringInMultipleDecorators() {
    AppendStringDecorator appendDecorator = new AppendStringDecorator(null, "Suffix");
    PrependStringDecorator prependDecorator = new PrependStringDecorator(appendDecorator, "prefix");
    StringDecorator stringDecorator = new EmptyStringDecorator(prependDecorator);
    assertEquals("prefixNameSuffix", stringDecorator.process("name"));
  }
}

File: UseCases/src/test/java/codes/seanhenry/mockgenerator/util/StringDecoratorTests.kt
package codes.seanhenry.mockgenerator.util

import junit.framework.TestCase
import kotlin.test.assertEquals

class StringDecoratorTests : TestCase() {

  internal inner class EmptyStringDecorator(decorator: StringDecorator?) : StringDecorator(decorator) {

    override fun decorate(string: String): String {
      return string
    }
  }

  fun testShouldProcessStringInDecorator() {
    val decorator = PrependStringDecorator(null, "prefix")
    val stringDecorator = EmptyStringDecorator(decorator)
    assertEquals("prefixName", stringDecorator.process("name"))
  }

  fun testShouldNotGiveStringToNullDecorator() {
    val stringDecorator = EmptyStringDecorator(null)
    assertEquals("name", stringDecorator.process("name"))
  }

  fun testShouldProcessStringInMultipleDecorators() {
    val appendDecorator = AppendStringDecorator(null, "Suffix")
    val prependDecorator = PrependStringDecorator(appendDecorator, "prefix")
    val stringDecorator = EmptyStringDecorator(prependDecorator)
    assertEquals("prefixNameSuffix", stringDecorator.process("name"))
  }
}
